samples/net/sockets/echo_service/prj.conf

```diff
@@ -0,0 +1,38 @@
+# General config
+# The async method used in the sample needs more stack for the workqueue
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1500
+CONFIG_POSIX_API=y
+
+# Networking config
+CONFIG_NETWORKING=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_TCP=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_NET_IPV4_MAPPING_TO_IPV6=y
+CONFIG_POSIX_MAX_FDS=10
+CONFIG_NET_MAX_CONN=5
+CONFIG_NET_SOCKETS_SERVICE=y
+CONFIG_NET_SOCKETS_POLL_MAX=20
+
+# Network driver config
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+# Network address config
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
```

Comment:
I only see IPv6 being used in the sample?

Reply:
The sample supports both IPv4 and IPv6 by turning off the V6ONLY socket option.